### PR TITLE
Removed mocking of tinySDF from tests and use the data from it

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist
This is basically #775 with the existing tests without mocking tinySDF.

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Write tests for all new functionality.
 - [x] Manually test the debug page.
 - [x] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [x] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Comment by HarelM** [comment]
I've tried to create a render test but the test passes with and without the changes in glyph manager so I'm not sure if this is a browser thing or not, but adding a test that passes even with these changes seems redundant.

**Review by birkskyum — APPROVED** [review]
Lgtm

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ### 🐞 Bug fixes
 
+- Throw error mismatched image size, when text-field is chinese (#718)
```
Comment (wipfli) [inline CHANGELOG.md:9]: Chinese should be replaced

Comment (wipfli) [reply CHANGELOG.md:9]: Maybe refer to CJK characters https://en.wikipedia.org/wiki/CJK_characters instead.

Comment (wipfli) [reply CHANGELOG.md:9]: Changelog entry on bug fix should start with "Fix"

Comment (wipfli) [reply CHANGELOG.md:9]: I suggest you use "Fix error mismatched image size for CJK characters (#718)"

**Review by wipfli — CHANGES_REQUESTED** [review]
Thanks @prozessor13 for contributing this bug fix and thanks @HarelM for updating the tests. Please fix the changelog, then this is approved from my side.

**Comment by HarelM** [comment]
Looking at the code and how it works I can't but wonder if this is a bug in TinySDF.
The reason I say that is that we create a tinySDF with hard coded parameters: font size 24, buffer 3.
If I need to guess this should always result in a 30 by 30 sized character.
Instead the results of the characters in this issues are 29 by 30 and 29 by 31, which seems like a rounding error (as if it should've been 30 by 30 but didn't quite make it).
@mourner do you want me to open an issue in TinySDF's repo on this?

**Review by wipfli — APPROVED** [review]
Thanks :)

**Comment by mourner** [comment]
> @mourner do you want me to open an issue in TinySDF's repo on this?

The behavior you're expecting was in v1. v2 returns the glyph of the size it's actually drawn, using text metrics, and it doesn't always align perfectly with `fontSize + 2 * buffer`, so it's working as designed I think.

**Comment by HarelM** [comment]
Ok, cool, thanks! We are using version 2 as of lately.

**Comment by prozessor13** [comment]
Thx @HarelM for finishing the Pull-Request!

**Pull request merged**